src/MemoryTracing/MemoryInfoProducer.cpp

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "MemoryTracing/MemoryInfoProducer.h"
+
+#include <chrono>
+#include <thread>
+
+#include "MemoryTracingUtils.h"
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/ThreadUtils.h"
+#include "capture.pb.h"
+
+namespace orbit_memory_tracing {
+
+using orbit_grpc_protos::SystemMemoryUsage;
+
+void MemoryInfoProducer::Start() {
+  exit_requested_ = false;
+  thread_ = new std::thread(&MemoryInfoProducer::Run, this);
+}
+
+void MemoryInfoProducer::Stop() {
+  exit_requested_ = true;
+  if (thread_ != nullptr && thread_->joinable()) {
+    thread_->join();
+  }
+  delete thread_;
```

Comment:
use unique_ptr<> for thread, and call reset() here.

Reply:
Would you prefer unique_ptr over std::optional?

Reply:
the class owns the thread here - std::optional won't delete it so I would say this should be std::unique_ptr<>
Rule of thumb: use unique_ptr to establish ownership of the object.

Reply:
Why won't `std::optional` delete it?

Reply:
Well, because I was thinking about std::optional<std::thread*>.
so, yes, you can use optional<thread> here.

Reply:
Done. Thanks for the suggestion! Changed to use `unique_ptr` for both `exit_requested_` and `thread_`.